internal/backup_selector.go

```diff
@@ -93,19 +93,24 @@ func (s UserDataBackupSelector) findBackupByUserData(userData interface{}, folde
 	return NewBackupInStorage(baseBackupFolder, foundBackupName, foundStorage)
 }
 
+type GenericMetadataInStorage struct {
+	GenericMetadata
+	StorageName string
+}
+
 // Search backups in storage using specified criteria
 func searchInMetadata(
 	criteria func(GenericMetadata) bool,
 	folder storage.Folder,
 	metaFetcher GenericMetaFetcher,
-) (metaByStorage map[string]GenericMetadata, err error) {
+) ([]GenericMetadataInStorage, error) {
```

Comment:
I found a bug that I made in some previous PR. This function could provide only a single backup metadata per storage, because I used `map` here.

This led to that any backup selection by metadata worked incorrectly - it couldn't find the desired backup even if it existed. And found oldest backups could not be actually the oldest ones.